### PR TITLE
Update gecco + migrate to nf-test

### PR DESCRIPTION
**Pull request by jasmezz**
- update gecco 0.9.8 -->  0.9.10 (uses pyrodigal 3 now)
- add stub test
- migrate to nf-test

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [x] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/gecco/run/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(
+                    process.out.genes +
+                    process.out.features +
+                    process.out.clusters +
```
Comment (jfy133) [inline modules/nf-core/gecco/run/tests/main.nf.test:33]: Whats with the `+`? I've not seen that before?

**Review by jfy133 — COMMENTED** [review]
There is no snapshot for the actual test, just the stub?

BUt otherwise looking good

File: modules/nf-core/gecco/run/tests/main.nf.test (hunk 1 above)
Comment (jasmezz) [reply modules/nf-core/gecco/run/tests/main.nf.test:33]: It's supposed to chain those outputs together and works. But thanks for the hint, I forgot to add the `.match()` which caused the test not to show up in the snapshot. Now it's fixed.

Comment (jasmezz) [reply modules/nf-core/gecco/run/tests/main.nf.test:33]: And the `+` is used by a few modules already, e.g. multiqc.

**Comment by jasmezz** [comment]
@nf-core-bot fix linting

File: modules/nf-core/gecco/run/tests/main.nf.test (hunk 1 above)
Comment (jfy133) [reply modules/nf-core/gecco/run/tests/main.nf.test:33]: Ok... I wonder what it doss

**Review by jfy133 — APPROVED** [review]

**Pull request merged**